install_config/persistent_storage/dynamically_provisioning_pvs.adoc

```diff
@@ -0,0 +1,160 @@
+
+[options="header"]
+|===
+
+|Storage Type |Provisioner Plug-in Name |Required Cloud Configuration
+
+|OpenStack Cinder
+|`kubernetes.io/cinder`
+|link:../../install_config/configuring_openstack.html[Configuring for OpenStack]
+
+|AWS Elastic Block Store (EBS)
+|`kubernetes.io/aws-ebs`
+|link:../../install_config/configuring_aws.html[Configuring for AWS]
+
+|GCE Persistent Disk (gcePD)
+|`kubernetes.io/gce-pd`
+|link:../../install_config/configuring_gce.html[Configuring for GCE]
+|===
+
+[IMPORTANT]
+====
+For any chosen provisioner plug-ins, configuration for the relevant cloud
+infrastructure must be set up as well, per *Required Cloud Configuration* in the
+above table.
+====
+
+Future provisioner plug-ins will include the many types of storage a single
+provider offers. AWS, for example, has several types of EBS volumes to offer,
+each with its own performance characteristics. They also have an NFS-like
+storage option. More provisioner plug-ins will be implemented for the supported
+storage types available in OpenShift.
+
+[[dynamic-pvs-configuring-storage-classes]]
+== Configuring Storage Classes
+
+Cluster administrators can configure their clusters to use dynamic provisioning
+for persistent volumes by defining _storage classes_ in the master configuration
+file (the *_/etc/origin/master/master-config.yml_* file by default). A storage
+class is a composite of the class name and the name of the provisioner plug-in,
+separated by a slash:
+
+----
+<storage_class_name>/<provisioner_plugin>
+----
+
+For example, for a storage class named *foo* mapping to an EBS volume:
+
+====
+----
+foo/kubernetes.io/aws-ebs
+----
+====
+
+Storage classes are associated with various quality of service levels, and each
+class should only be associated with one provisioner:
+
+.Configuring Storage Classes in the Master Configuration File
+====
+[source,yaml]
+----
+kubernetesMasterConfig:
+  apiLevels:
+  - v1beta3
+  - v1
+  apiServerArguments: null
+  controllerArguments:
+    "storage-class": <1>
+      - "foo/kubernetes.io/aws-ebs"
+      - "bar/kubernetes.io/cinder"
+      - "baz/kubernetes.io/gce-pd"
+----
+<1> Add a `*"storage-class"*` section to
+`*kubernetesMasterConfig.controllerArguments*` and list the desired storage
+classes.
+====
+
+Restart the master service for the changes to take effect:
+
+----
+# systemctl restart atomic-openshift-master
+----
+
+[[dynamic-pvs-requesting-storage]]
+== Requesting Dynamically Provisioned Storage
+
+Users can request dynamically provisioned storage by including a storage class
+annotation in their persistent volume claim. The value of the annotation must be
+one of the storage class names as configured by the cluster administrator:
+
+.Persistent Volume Claim Requesting Dynamic Storage
+====
+[source,yaml]
+----
+kind: "PersistentVolumeClaim"
+apiVersion: "v1"
+metadata:
+  name: "claim1"
+  annotations:
+    volume.experimental.kubernetes.io/quality-of-service: "foo" <1>
```

Comment:
It should be noted that the value of the annotation is not meaningful at this time.  The presence of the annotation (with any arbitrary value) triggers provision with the single implied provisioner per cloud.